syntax = "proto3";
package payload.manager.v1;

import "payload/manager/v1/placement.proto";
import "payload/manager/v1/id.proto";
import "payload/manager/v1/types.proto";
import "payload/manager/v1/policy.proto";

/*
  Allocate reserves placement but payload is not visible
  until CommitPayload succeeds.
*/
message AllocatePayloadRequest {
  uint64 size_bytes = 1;
  Tier preferred_tier = 2;
  uint64 ttl_ms = 3;
  bool persist = 4;
  EvictionPolicy eviction_policy = 5;
}

message AllocatePayloadResponse {
  PayloadDescriptor payload_descriptor = 1;
}

message CommitPayloadRequest {
  PayloadID id = 1;
}

message CommitPayloadResponse {
  PayloadDescriptor payload_descriptor = 1;
}

/*
  force=true invalidates all active leases immediately.
*/
message DeleteRequest {
  PayloadID id = 1;
  bool force = 2;
}
